storage/grpc_client.go

```diff
@@ -550,7 +550,7 @@ func (c *grpcStorageClient) UpdateObject(ctx context.Context, params *updateObje
 	if uattrs.TemporaryHold != nil {
 		fieldMask.Paths = append(fieldMask.Paths, "temporary_hold")
 	}
-	if uattrs.ContentType != nil {
+	if uattrs.ContentType != nil || params.forceEmptyContentType {
```

Comment:
This looks like the wrong place to fix -- you need to make the change here: https://github.com/googleapis/google-cloud-go/blob/cc83515d0c837c8b1596a97b6f09d519a0f75f72/storage/grpc_client.go#L1784

I think if params.forceEmptyContentType is set then we can skip the sniffing step on the following line.

Reply:
Thanks, I ran the integration test on the latest changes, and now it passes with gRPC.